core/src/main/kotlin/no/ntnu/beardblaster/spell/SpellExecutor.kt

```diff
@@ -15,35 +15,43 @@ class SpellExecutor {
     private val mitigationSpells: MutableList<MitigationSpell> = ArrayList()
 
     fun addSpell(spell: SpellAction, turn: Int) {
-        LOG.info { "Adding spell ${spell.spell.spellName} cast by ${spell.caster} - turn $turn" }
-        if(spellHistory.containsKey(turn)) {
-            spellHistory[turn]?.add(spell)
-        }
-        spellHistory.putIfAbsent(turn, mutableListOf(spell))
+        if(isSpellValidForTurn(spell, turn)) {
+            LOG.info { "Adding spell ${spell.spell.spellName} cast by ${spell.caster} - turn $turn" }
+            if(spellHistory.containsKey(turn)) {
+                spellHistory[turn]?.add(spell)
+            }
+            spellHistory.putIfAbsent(turn, mutableListOf(spell))
 
-        if (spell.spell.spellHealing > 0) {
-            mitigationSpells.add(HealingOverTime(spell.caster, turn, spell.spell.duration, spell.spell.spellHealing))
-        }
+            if (spell.spell.spellHealing > 0) {
+                mitigationSpells.add(HealingOverTime(spell.caster, turn, spell.spell.duration, spell.spell.spellHealing))
+            }
 
-        if (spell.spell.spellMitigation > 0) {
-            mitigationSpells.add(
-                DamageReduction(
-                    spell.caster,
-                    turn + 1, // Damage reduction spells should be activated 1 turn after current
-                    spell.spell.duration,
-                    spell.spell.spellMitigation
+            if (spell.spell.spellMitigation > 0) {
+                mitigationSpells.add(
+                    DamageReduction(
+                        spell.caster,
+                        turn,
```

Comment:
Previously turn was incremented by 1 here - but this had already been accounted for in DamageReduction (leading to damage reduction being activated 1 turn in advance)